src/predictor/gpu_predictor.cu

```diff
@@ -481,20 +575,44 @@ void ExtractPaths(dh::device_vector<gpu_treeshap::PathElement>* paths,
       double child_cover = d_stats[child_idx].sum_hess;
       double parent_cover = d_stats[parent_idx].sum_hess;
       double zero_fraction = child_cover / parent_cover;
-      auto parent = d_nodes[parent_idx];
+      auto parent = tree.d_tree[child.Parent()];
+
       bool is_left_path = (tree_offset + parent.LeftChild()) == child_idx;
       bool is_missing_path = (!parent.DefaultLeft() && !is_left_path) ||
                              (parent.DefaultLeft() && is_left_path);
-      float lower_bound = is_left_path ? -inf : parent.SplitCond();
-      float upper_bound = is_left_path ? parent.SplitCond() : inf;
-      d_paths[output_position--] = {
-          idx,         parent.SplitIndex(), group,         lower_bound,
-          upper_bound, is_missing_path,     zero_fraction, v};
+
+      float lower_bound = -inf;
+      float upper_bound = inf;
+      common::CatBitField bits;
+      if (common::IsCat(tree.cats.split_type, child.Parent())) {
+        auto path_cats = d_path_categories.subspan(max_cat * output_position, max_cat);
+        size_t size = tree.cats.node_ptr[child.Parent()].size;
+        auto node_cats = tree.cats.categories.subspan(tree.cats.node_ptr[child.Parent()].beg, size);
+        SPAN_CHECK(path_cats.size() >= node_cats.size());
+        if (is_left_path) {
+          for (size_t i = 0; i < node_cats.size(); ++i) {
+            path_cats[i] = ~node_cats[i];
```

Comment:
`path_cats[i] = is_left_path ? ~node_cats[i] : node_cats[i];`?

Reply:
So in the bitset and xgboost predict, values that are set to 1 go to right, 0 go to left, and each bit corresponds to a category.   But when we construct the shap path, there's no left or right in the split condition, only "whether should we go down this path".  As a result, for paths that go to right node, we keep the bits unchanged, but for the paths that go to the left, we flip the bits.

Reply:
Done.